src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        List<CustomerSalary> customerSalaryList = new ArrayList<>();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+        for (String record : data) {
+            String[] recordArr = record.split(" ");
+            LocalDate currentDate = LocalDate.parse(recordArr[0], formatter);
+            LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+            LocalDate toDate = LocalDate.parse(dateTo, formatter);
+            if ((currentDate.isAfter(fromDate) || currentDate.equals(fromDate))
+                    && (currentDate.isBefore(toDate) || currentDate.equals(toDate))) {
+                customerSalaryList.add(new CustomerSalary(currentDate, recordArr[1],
+                        Integer.valueOf(recordArr[2]), Integer.valueOf(recordArr[3])));
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#any-magic-numbers-should-be-constants): Any magic numbers should be constants. Lets make numbers 0, 1, 2, 3 as constants with informative names

Reply:
Done